api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -92,6 +92,11 @@ message ExtAuthz {
   // or cannot be reached. The default status is HTTP 403 Forbidden.
   type.v3.HttpStatus status_on_error = 7;
 
+  // Sets the HTTP status that is returned to the client when the authorization server returns a
+  // response that is rejected by the filter / client. The default status is HTTP 500 Internal
+  // Server Error.
+  type.v3.HttpStatus status_on_rejected = 24;
```

Comment:
nit:  change name to `status_on_validation_failure`

---

Sorry, seems I am involved because this API change. But rather than a runtime guard that will finally be deprecated and removed, I prefer an explicitly API to enable the headers validation would better.
I believe most users will use a trusted auth server and even some users may doesn't want to reject invalid headers (even HCM could accept invalid header name with '_' by setting).

/wait-any

Reply:
Thanks for that insight, that makes sense.

Reply:
Agreed; this makes more sense as config than a runtime guard.